### PR TITLE
Add a new endpoint where tool usage events can be logged from steps

### PR DESCRIPTION
**Pull request by mateherber**
We'd like to track the used / installed steps during a workflow so we
can prepare our VMs better for real life usage.

* Added new endpoint
* Added new event structure
* Added test cases

[STEP-381]

[STEP-381]: https://bitrise.atlassian.net/browse/STEP-381

### CODE REVIEW

**Review by sylank — APPROVED** [review]
Could you insert the card ID into the PR and into the further commit message? Otherwise LGTM.

File: service/tools_post.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	}
+
+	if toolAnalytics.BuildSlug == "" {
+		return httpresponse.RespondWithBadRequestError(w, "Invalid request body, please provide build_slug")
+	}
+	if len(toolAnalytics.ToolUsage) < 1 {
+		return httpresponse.RespondWithBadRequestError(w, "Invalid request body, please provide at least one tool_usage")
+	}
+	for _, toolUsage := range toolAnalytics.ToolUsage {
+		if toolUsage.Name == "" {
```
Comment (adborbas) [inline service/tools_post.go:26]: What about the other properties `Version` and `FreshInstall`?

Comment (mateherber) [reply service/tools_post.go:26]: The default value is OK for them

Comment (adborbas) [reply service/tools_post.go:26]: Ok, I see, I agree with `Version` being empty. But are we correctly assuming that if we cannot determine if a tool is freshly installed, we say it is updated?

**Review by sylank — APPROVED** [review]

File: service/tools_post.go (hunk 1 above)
Comment (mateherber) [reply service/tools_post.go:26]: Yes. Our tool will be filling the params right. Other users of the endpoint will have to fill the field properly. I think false is a safe default here.

**Review by adborbas — APPROVED** [review]

**Pull request merged**